### DISCUSSION TITLE
Hide tabline when number of tabs is lower than two

### DISCUSSION THREAD

**Discussion by simonmandlik** [nvim-lualine/lualine.nvim#845] [Q&A]
Hi,

in neovim 0.7 (and the latest version of lualine), this stopped working for me

```
autocmd OptionSet showtabline :set showtabline=1
```

Judging by https://github.com/nvim-lualine/lualine.nvim/issues/395, I'm not alone, but since the issue is closed, what is the preferrable way of achieving it? Thanks!

**Comment by fitrh** [comment]
Maybe you can do something like this

```lua
vim.api.nvim_create_autocmd({ "TabNew", "TabClosed" }, {
  group = vim.api.nvim_create_augroup("LualineTab", {}),
  callback = function()
    local show = #vim.api.nvim_list_tabpages() > 1
    vim.o.showtabline = show and 1 or 0
    require("lualine").hide({
      place = { "tabline" },
      unhide = show,
    })
  end,
})
```

The problem is, how to hide the tabline on the first launch?
It seems `VimEnter` did not work here

**Reply by simonmandlik** [reply]
Thank you, that would work, but it seems quite a lot of code for such simple setting. Given that `showtabline` is a standard vim setting, shouldn't code like this be a part of this plugin?

**Reply by fitrh** [reply]
I agree, by respecting user `'showtabline'`, it could be done simpler like this

```diff
diff --git a/lua/lualine.lua b/lua/lualine.lua
index a7db85f..00bee0c 100644
--- a/lua/lualine.lua
+++ b/lua/lualine.lua
@@ -469,7 +469,7 @@ local function set_tabline(hide)
       "call v:lua.require'lualine'.refresh({'kind': 'tabpage', 'place': ['tabline'], 'trigger': 'autocmd'})",
       'lualine_tal_refresh'
     )
-    modules.nvim_opts.set('showtabline', 2, { global = true })
+    modules.nvim_opts.set('showtabline', vim.o.showtabline, { global = true })
     timers.halt_tal_refresh = false
   else
     modules.nvim_opts.restore('tabline', { global = true })
```

What do you think @shadmansaleh? I can't find any reference behind the decision
to set `'showtabline'` to 2 by default

**Reply by simonmandlik** [reply]
@shadmansaleh please?

**Reply by scienceplease** [reply]
> Maybe you can do something like this
> 
> ```lua
> vim.api.nvim_create_autocmd({ "TabNew", "TabClosed" }, {
>   group = vim.api.nvim_create_augroup("LualineTab", {}),
>   callback = function()
>     local show = #vim.api.nvim_list_tabpages() > 1
>     vim.o.showtabline = show and 1 or 0
>     require("lualine").hide({
>       place = { "tabline" },
>       unhide = show,
>     })
>   end,
> })
> ```
> 
> The problem is, how to hide the tabline on the first launch? It seems `VimEnter` did not work here

A simpler hack would be to just create a dummy function-defined component in the tabline that sets `vim.o.showtabline` to your preferred setting. It's rather silly, but I couldn't find a better way to hook into lualine's API to detect when the tabline is set. Hopefully the actual fix will get merged soon. 
```lua
  tabline = {
      lualine_c = {
          function() vim.o.showtabline = 1; return '' end
      },
```

**Comment by sleeptightAnsiC** [comment]
Here it is:
https://github.com/nvim-lualine/lualine.nvim/blob/0a5a66803c7407767b799067986b4dc3036e1983/README.md?plain=1#L433-L435
```lua
{
	'tabs',
	cond = function() return 2 <= #vim.api.nvim_list_tabpages() end,
},
```
Using this does the trick on my side.

**Reply by alexspa** [reply]
If I combine it with `vim.opt.showtabline = 1` the margin required by the tabline also gets set as expected

```lua
{
'tabs',
cond = function()
    vim.opt.showtabline = 1
    return 2 <= #vim.api.nvim_list_tabpages()
end,
}
```
